src/module.c

```diff
@@ -1120,6 +1120,26 @@ void swapDbEventCallback(RedisModuleCtx *ctx, RedisModuleEvent e, uint64_t sub,
     }
 }
 
+int persistence_in_progress = 0;
+
+void persistCallback(RedisModuleCtx *ctx, RedisModuleEvent eid, uint64_t subevent, void *data) {
+    if (memcmp(&eid, &RedisModuleEvent_Persistence, sizeof(eid)) != 0) {
```

Comment:
I do not think this is needed, you register this callback on the Persistence event only.

Reply:
I think it's more safe to make sure it's the type of event, don't you think?